### PR TITLE
heatseeker: fix finding stty

### PR DESCRIPTION
**Pull request by michaelpj**
Otherwise this crashes at runtime.

I've opened https://github.com/rschmitt/heatseeker/issues/42 upstream.

@bhipple this is the fix from https://github.com/NixOS/nixpkgs/pull/80539 that's still needed.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/heatseeker/default.nix [hunk 1]
```diff
@@ -13,6 +13,13 @@ rustPlatform.buildRustPackage rec {
 
   cargoSha256 = "0jnlcm7v29m4nc318qgf7r7jvs80s7n04fw83imm506vwr9rxbx9";
 
+  # https://github.com/rschmitt/heatseeker/issues/42
+  # I've suggested using `/usr/bin/env stty`, but doing that isn't quite as simple
+  # as a substitution, and this works since we have the path to coreutils stty.
+  patchPhase = ''
+    substituteInPlace src/screen/unix.rs --replace "/bin/stty" "${coreutils}/bin/stty"
```
Comment (cole-h) [inline pkgs/tools/misc/heatseeker/default.nix:20]: You need to add `coreutils` as an argument (on the first line) to make `ofborg` happy again.

**Comment by michaelpj** [comment]
Rats, I pulled out this patch from another commit and missed a bit, sorry!

**Comment by michaelpj** [comment]
Fixed!

**Comment by michaelpj** [comment]
Gentle ping.

**Review by cole-h — APPROVED** [review]
Diff LGTM. Binary works, tested with `ls | hs`.

```
[2 built, 2 copied (8.8 MiB), 8.6 MiB DL]
https://github.com/NixOS/nixpkgs/pull/84662
1 package built:
heatseeker
```

**Comment by michaelpj** [comment]
ping @bhipple

**Comment by timokau** [comment]
Clean improvement, well-executed with a reference to an upstream issue :+1: 

Sorry you had to wait so long.

**Pull request merged**